vmdb/app/models/ems_refresh/parsers/foreman.rb

```diff
@@ -0,0 +1,88 @@
+module EmsRefresh::Parsers
+  class Foreman
+    def self.ems_inv_to_hashes(inv)
+      new.ems_inv_to_hashes(inv)
+    end
+
+    def ems_inv_to_hashes(inv)
+      result = {}
+      ids = {}
+      result[:customization_scripts] = customization_scripts_inv_to_hash(inv[:media], inv[:ptables])
+      add_ids(ids, result[:customization_scripts])
+      result[:operating_system_flavors] = operating_system_flavors_inv_to_hashes(inv[:operating_systems], ids)
+      add_ids(ids, result[:operating_system_flavors])
+      result[:configuration_profiles] = configuration_profile_inv_to_hashes(inv[:hostgroups], ids)
+      add_ids(ids, result[:configuration_profiles])
+      result[:configured_systems] = configured_system_inv_to_hashes(inv[:hosts], ids)
+      result
+    end
+
+    def media_inv_to_hashes(media)
+      media.map do |m|
+        {
+          "manager_ref"  => "medium:#{m["id"]}",
```

Comment:
this is tricky one.  You *could* make the key on type and manager_ref in the save_inventory.  We've had to do mutli-key checks in other places, and that should work fine.

Otherwise, this is fine.  It makes it kind of weird because there are patterns elsewhere to access native objects by ems_ref, so you would need to do something different and parse the ems_ref before using it.  For example, there is a method with_provider_object that could be defined on CustomizationScriptMedium, and internally it would use ems_ref directly.